src/core/qgsvectorlayer.cpp

```diff
@@ -157,6 +158,7 @@ QgsVectorLayer::QgsVectorLayer( const QString &vectorLayerPath,
   mGeometryOptions = qgis::make_unique<QgsGeometryOptions>();
   mActions = new QgsActionManager( this );
   mConditionalStyles = new QgsConditionalLayerStyles();
+  mStoredExpressionManager = new QgsStoredExpressionManager();
```

Comment:
If this IS a QObject, it should be parented to the layer and let Qt handle ownership (also correct transferal if the layer is moved between threads)